### PR TITLE
Use try with resources for file system

### PR DESCRIPTION
**Pull request by asaarilahti**
Related to #332 - close file system also in case an exception is thrown. Unlikely to be an actual fix, but at least should rule out earlier failures as a cause.

Also switch from a visitor to a stream.

### CODE REVIEW

**Comment by cberg-zalando** [comment]
Any reason for the switch from visitor to stream?

**Comment by asaarilahti** [comment]
> Any reason for the switch from visitor to stream?

Nothing major. I think the visitor is not that common nowadays, so streams might make the code easier to read.

File: src/main/java/org/zalando/intellij/swagger/file/SwaggerUiCreator.java [hunk 1]
```diff
@@ -61,34 +60,28 @@ private void setSwaggerConfigurationValues(
...
-            return FileVisitResult.CONTINUE;
-          }
-        });
-
-    fileSystem.close();
+    try (FileSystem fileSystem = FileSystems.newFileSystem(resource, Map.of())) {
+
+      final Path jarPath = fileSystem.getPath(resourcePath);
+
+      try (Stream<Path> stream = Files.walk(jarPath)) {
```
Comment (cberg-zalando) [inline src/main/java/org/zalando/intellij/swagger/file/SwaggerUiCreator.java:67]:
```suggestion
      try (Stream<Path> stream = Files.walk(fileSystem.getPath(resourcePath))) {
```

Comment (cberg-zalando) [reply src/main/java/org/zalando/intellij/swagger/file/SwaggerUiCreator.java:67]: You don't use the variable later, so why actually assign it at all?!

Comment (asaarilahti) [reply src/main/java/org/zalando/intellij/swagger/file/SwaggerUiCreator.java:67]: > You don't use the variable later, so why actually assign it at all?!

Would be nice, but it's actually used (in lines 72 and 77).

Comment (cberg-zalando) [reply src/main/java/org/zalando/intellij/swagger/file/SwaggerUiCreator.java:67]: So much for my eyesight. :smirk:

**Review by cberg-zalando — APPROVED** [review]

**Pull request merged**